cmd/sonobuoy/app/version.go

```diff
@@ -20,20 +20,76 @@ import (
 	"fmt"
 
 	"github.com/heptio/sonobuoy/pkg/buildinfo"
+	"github.com/heptio/sonobuoy/pkg/errlog"
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 )
 
-func init() {
-	RootCmd.AddCommand(versionCmd)
+type versionFlags struct {
+	kubecfg Kubeconfig
 }
 
-var versionCmd = &cobra.Command{
-	Use:   "version",
-	Short: "Print sonobuoy version",
-	Run:   runVersion,
-	Args:  cobra.ExactArgs(0),
+var versionflags versionFlags
+
+func init() {
+	cmd := &cobra.Command{
+		Use:   "version",
+		Short: "Print sonobuoy version",
+		Run:   runVersion,
+		Args:  cobra.ExactArgs(0),
+	}
+
+	AddKubeconfigFlag(&versionflags.kubecfg, cmd.Flags())
+
+	RootCmd.AddCommand(cmd)
 }
 
 func runVersion(cmd *cobra.Command, args []string) {
-	fmt.Println(buildinfo.Version)
+
+	fmt.Println(fmt.Sprintf("Sonobuoy Version: %s", buildinfo.Version))
+	fmt.Println(fmt.Sprintf("MinimumKubeVersion: %s", buildinfo.MinimumKubeVersion))
+	fmt.Println(fmt.Sprintf("MaximumKubeVersion: %s", buildinfo.MaximumKubeVersion))
+	fmt.Println(fmt.Sprintf("GitSHA: %s", buildinfo.GitSHA))
+
+	// Get Kubernetes version, this is last so that the regular version information
+	// will be shown even if the API server cannot be contacted and throws an error
+	apiVersion, skipk8sCheck := getK8Sversion()
+	if !skipk8sCheck {
+		fmt.Println("API Version: ", apiVersion)
+	} else {
+		fmt.Println("API Version check skipped due to missing `--kubeconfig` or other error")
```

Comment:
I'd much rather expose the issue: if I provided a config but still got this message I'd be pretty frustrated that there wasn't more to go off of. Any reason you wanted to hide them?